rx_signal_processing/dsp.py

```diff
@@ -221,100 +222,123 @@ def correlations_from_samples(beamformed_samples_1, beamformed_samples_2, output
 
             # [num_range_gates, 1, 1]
             # [1, num_lags, 2]
-            samples_for_all_range_lags = (range_off[...,np.newaxis,np.newaxis] +
-                                            lag_pulses_as_samples[np.newaxis,:,:])
-
+            samples_for_all_range_lags = (range_off[..., np.newaxis, np.newaxis] +
+                                          lag_pulses_as_samples[np.newaxis, :, :])
 
             # [num_range_gates, num_lags, 2]
-            row = samples_for_all_range_lags[...,1].astype(np.int32)
+            row = samples_for_all_range_lags[..., 1].astype(np.int32)
 
             # [num_range_gates, num_lags, 2]
-            column = samples_for_all_range_lags[...,0].astype(np.int32)
+            column = samples_for_all_range_lags[..., 0].astype(np.int32)
 
-            values_for_slice = correlated[s['slice_num'],:,row,column]
+            values_for_slice = correlated[s['slice_num'], :, row, column]
 
             # [num_range_gates, num_lags, num_beams]
-            values_for_slice = np.einsum('ijk->kij', values_for_slice)
+            values_for_slice = np.einsum('ijk,j->kij', values_for_slice, s['lag_phase_offsets'])
```

Comment:
This is getting added in a different pull request, might be better not to add it in here. Merge conflicts and all